scrunch/datasets.py

```diff
@@ -761,15 +806,19 @@ def graph(self, _):
     def get(self):
         # Returns the synchronized hierarchical order graph.
         if self._sync:
-            ds_state = self.ds.resource.session.get(
-                self.ds.resource.self + 'state/'
-            ).payload
-            if self._revision is None:
-                self._revision = ds_state.body.revision
-            elif self._revision != ds_state.body.revision:
-                # There's a new dataset revision. Reload the
-                # hierarchical order.
-                self._revision = ds_state.body.revision
+            if hasattr(self.resource, 'variables'):
+                ds_state = self.resource.session.get(
```

Comment:
If we need different or extra rules for the variables order and the dataset order we should do this inside subclasses instead of cases here. Most of the operations should live in the `Order` class and if necessary subclass `DatasetVariableOrder`, `ProjectDatasetsOrder`. Hinting for attributes should be considered a bad practice